### DISCUSSION TITLE
Honour extent environment settings

### DISCUSSION THREAD

**Discussion by Hornbydd** [opengeos/WhiteboxTools-ArcGIS#52] [Ideas]
What would be a really useful improvement to the toolbox is that it honours the processing extent, an environment setting regularly used to limit the area you wish to process. I assume this toolbox does not use processing extent simply because WBT does not accept it as a parameter? May be you could discuss this with John?  Such a setting is essential when the input raster is very large and you need only process a smaller area. Yes one could go to the effort and clip it out but setting the processing extent is so much easier and slicker.

**Comment by giswqs** [comment]
I agree that this would be a very useful feature. One issue is that there are many non-raster tools in whitebox that the extent settings can't be applied to. It would be challenging to implement this in the WBT frontends.

**Reply by Hornbydd** [reply]
I appreciate it was a lot to ask for! A possible solution is to make WBT tools that could use an extent, accept it as an additional parameter rather than an environment setting? So your toolbox still ignores environment settings but then for specific tools one could feed in an optional extent parameter? Just an idea.

**Reply by giswqs** [reply]
This would be better for the WBT backend to implement. It would be hard for the frontend to loop through 500+ tools and keep track of which tools can accept the optional extent parameter. 

One potential workaround is to check if an extent parameter is set. If yes, run the arcpy extract_by_mask tool to clip the image first before passing it to WBT tools. Nevertheless, it would still need to figure out which input(s) is a raster that need to be clipped. It maybe feaisble to do this for specific tools, but it can get quite complicated to check this for all tools with very different input parameters. Note that the entire Arc WBT toolbox is generated automatically by this [automation.py](https://github.com/opengeos/WhiteboxTools-ArcGIS/blob/master/WBT/PRE/automation.py), so whatever parameters we add need to be applicable to all tools. If not, a checking mechanism must exsit under the hood to ignore it before calling the WBT backend.
